src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -1355,6 +1377,111 @@ def _deny_request(
+        )
+    replies_for_privacy_request = PreApprovalWebhookReply.filter(
+        db=db,
+        conditions=(PreApprovalWebhookReply.privacy_request_id == privacy_request_id),
+    ).all()
+
+    # Check if not all replies have been received
+    if len(replies_for_privacy_request) < len(all_webhooks):
+        # Subtlety here is that if a new webhook is configured between when the privacy request was created and when
+        # the last webhook replied, then the privacy request can never be automatically approved. It must instead be
+        # manually approved in the Admin UI.
+        return
+    # Check if all replies are true
+    if not all(reply.is_eligible is True for reply in replies_for_privacy_request):
+        return
+
+    def _approve_request(privacy_request: PrivacyRequest) -> None:
+        """Method for how to process requests - approved"""
+        now = datetime.utcnow()
+        privacy_request.status = PrivacyRequestStatus.approved
+        privacy_request.reviewed_at = now
+        if privacy_request.custom_fields:  # type: ignore[attr-defined]
+            privacy_request.custom_privacy_request_fields_approved_at = now
+        privacy_request.save(db=db)
+        AuditLog.create(
+            db=db,
+            data={
+                "user_id": None,
+                "privacy_request_id": privacy_request.id,
+                "action": AuditLogAction.approved,
+                "message": "",
+                "webhook_id": webhook.id,
+            },
+        )
+        if config_proxy.notifications.send_request_review_notification:
+            _send_privacy_request_review_message_to_user(
+                action_type=MessagingActionType.PRIVACY_REQUEST_REVIEW_APPROVE,
+                identity_data=privacy_request.get_cached_identity_data(),
+                rejection_reason=None,
+                service_type=config_proxy.notifications.notification_service_type,
+            )
+
+        queue_privacy_request(privacy_request_id=privacy_request.id)
+
+    review_privacy_request(
+        db=db,
+        request_ids=[privacy_request_id],
+        process_request_function=_approve_request,
+    )
+
+
+@router.post(
+    PRIVACY_REQUEST_PRE_APPROVE_NOT_ELIGIBLE,
+    status_code=HTTP_200_OK,
+)
+def mark_privacy_request_pre_approve_not_eligible(
+    privacy_request_id: str,
+    *,
+    db: Session = Depends(deps.get_db),
+    webhook: PreApprovalWebhook = Security(
+        verify_callback_oauth_pre_approval_webhook, scopes=[PRIVACY_REQUEST_REVIEW]
+    ),
+) -> None:
+    """Marks privacy request as not eligible for automatic approval, regardless of what other webhook responses we receive"""
+    PreApprovalWebhookReply.create(
+        db=db,
+        data={
+            "webhook_id": webhook.id,
+            "privacy_request_id": privacy_request_id,
+            "is_eligible": False,
+        },
+    )
```

Comment:
I also might only save this if the Privacy Request is in "pending" status.  This could cause confusion otherwise.